context/app/static/js/components/Search/SearchWrapper.jsx

```diff
@@ -49,9 +53,40 @@ function SearchWrapper(props) {
     };
   }, [searchkit, setSearchHitsCount]);
 
+  const [message, setMessage] = useState(null);
+  const { elasticsearchEndpoint, nexusToken } = useContext(AppContext);
+  searchkit.setQueryProcessor((query) => {
+    // setQueryProcessor is typically used to change the query,
+    // but we're just using it as a place to call hooks.
+    // Make sure that every exit point from this function return the query.
+    const musts = query?.post_filter?.bool?.must;
+    if (!musts) {
+      setMessage(null);
+      return query;
+    }
+    const labels = {
+      'ancestor_ids.keyword': 'Derived from',
+      'descendant_ids.keyword': 'Ancestor of',
+    };
+    const provMusts = musts.filter(
+      (must) => 'term' in must && Object.keys(labels).some((labelKey) => labelKey in must.term),
+    );
+    setMessage(
+      provMusts.map((must) => {
+        return Object.entries(must.term).map(([k, v]) => (
+          <LookupEntity uuid={v} elasticsearchEndpoint={elasticsearchEndpoint} nexusToken={nexusToken}>
```

Comment:
This is the only place we're using `LookupEntity` would one of our hooks fit instead?